lib/Dialect/FIRRTL/LowerToRTL.cpp

```diff
@@ -440,6 +461,33 @@ LogicalResult FIRRTLLowering::lowerBinOp(Operation *op) {
   return setLoweringTo<ResultOpType>(op, lhs, rhs);
 }
 
+/// lowerCmpOp extends each operand to the destination type, then performs the
+/// specified binary operator.
+template <typename ResultOpSignedType, typename ResultOpUnsignedType>
+LogicalResult FIRRTLLowering::lowerCmpOp(Operation *op, bool flip) {
+  // Extend the two operands to match the destination type.
+  Type resultType = builder->getIntegerType(1);
+  auto lhs = getLoweredValue(op->getOperand(0));
+  auto rhs = getLoweredValue(op->getOperand(1));
```

Comment:
Does firrtl allow mixed type operands to comparisons (I didn't check, but it probably does...) if so you need to extend to common types here.